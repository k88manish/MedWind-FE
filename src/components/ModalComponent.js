import React from "react";
import { Modal, Button } from "react-bootstrap";

const ModalComponent = ({ show, handleClose, message = "", title }) => (
  <Modal show={show} onHide={handleClose}>
    <Modal.Header closeButton>
      <Modal.Title>{title}</Modal.Title>
    </Modal.Header>
    <Modal.Body>{message}</Modal.Body>
    <Modal.Footer>
      <Button className="close-btn" variant="secondary" onClick={handleClose}>
        Close
      </Button>
    </Modal.Footer>
  </Modal>
);

export default ModalComponent;
