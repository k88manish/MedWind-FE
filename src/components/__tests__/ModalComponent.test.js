import React from "react";
import { shallow } from "enzyme";
import ModalComponent from "../ModalComponent";

describe("ModelComponent testing", () => {
  const handleCloseMock = jest.fn();
  const props = {
    show: true,
    handleClose: handleCloseMock,
    message: "test",
    title: "test title"
  };
  const component = shallow(<ModalComponent {...props} />);

  it("should render properly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should call handleClose function when", () => {
    const btn = component.find(".close-btn");
    btn.simulate("click");
    expect(handleCloseMock).toHaveBeenCalledTimes(1);
  });
});
