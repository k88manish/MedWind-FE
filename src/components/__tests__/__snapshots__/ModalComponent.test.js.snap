// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelComponent testing should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalComponent
    handleClose={[MockFunction]}
    message="test"
    show={true}
    title="test title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          closeButton={true}
          closeLabel="Close"
        >
          <ForwardRef>
            test title
          </ForwardRef>
        </ForwardRef>,
        <ForwardRef>
          test
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            active={false}
            className="close-btn"
            disabled={false}
            onClick={[MockFunction]}
            type="button"
            variant="secondary"
          >
            Close
          </ForwardRef>
        </ForwardRef>,
      ],
      "onHide": [MockFunction],
      "show": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef>
            test title
          </ForwardRef>,
          "closeButton": true,
          "closeLabel": "Close",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test title",
          },
          "ref": null,
          "rendered": "test title",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalTitle",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            active={false}
            className="close-btn"
            disabled={false}
            onClick={[MockFunction]}
            type="button"
            variant="secondary"
          >
            Close
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": "Close",
            "className": "close-btn",
            "disabled": false,
            "onClick": [MockFunction],
            "type": "button",
            "variant": "secondary",
          },
          "ref": null,
          "rendered": "Close",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "active": false,
              "disabled": false,
              "type": "button",
              "variant": "primary",
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalFooter",
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "BACKDROP_TRANSITION_DURATION": 150,
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalBody",
        "render": [Function],
      },
      "Dialog": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "ModalDialog",
        "render": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalFooter",
        "render": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "closeButton": false,
          "closeLabel": "Close",
        },
        "displayName": "ModalHeader",
        "render": [Function],
      },
      "TRANSITION_DURATION": 300,
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "ModalTitle",
        "render": [Function],
      },
      "defaultProps": undefined,
      "displayName": "Bootstrap(Modal)",
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            closeButton={true}
            closeLabel="Close"
          >
            <ForwardRef>
              test title
            </ForwardRef>
          </ForwardRef>,
          <ForwardRef>
            test
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              active={false}
              className="close-btn"
              disabled={false}
              onClick={[MockFunction]}
              type="button"
              variant="secondary"
            >
              Close
            </ForwardRef>
          </ForwardRef>,
        ],
        "onHide": [MockFunction],
        "show": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef>
              test title
            </ForwardRef>,
            "closeButton": true,
            "closeLabel": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test title",
            },
            "ref": null,
            "rendered": "test title",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "ModalTitle",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "closeButton": false,
              "closeLabel": "Close",
            },
            "displayName": "ModalHeader",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalBody",
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              active={false}
              className="close-btn"
              disabled={false}
              onClick={[MockFunction]}
              type="button"
              variant="secondary"
            >
              Close
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "Close",
              "className": "close-btn",
              "disabled": false,
              "onClick": [MockFunction],
              "type": "button",
              "variant": "secondary",
            },
            "ref": null,
            "rendered": "Close",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "ModalFooter",
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "BACKDROP_TRANSITION_DURATION": 150,
        "Body": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalBody",
          "render": [Function],
        },
        "Dialog": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        },
        "Footer": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalFooter",
          "render": [Function],
        },
        "Header": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "closeButton": false,
            "closeLabel": "Close",
          },
          "displayName": "ModalHeader",
          "render": [Function],
        },
        "TRANSITION_DURATION": 300,
        "Title": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "ModalTitle",
          "render": [Function],
        },
        "defaultProps": undefined,
        "displayName": "Bootstrap(Modal)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
